pkg/vtpm/src/server.cpp

```diff
@@ -56,35 +212,41 @@ int readMessage(int sock, google::protobuf::uint32 size)
     CodedStrmInput.PopLimit(msgLimit);
 
     cout << "Received command is " << request.command() << std::endl;
+    if (sanitizeCmdRequest(sock, request, response) != 0) {
+         return -1;
+    }
     for (int i=0; i < request.inputfiles_size(); i++) {
         const eve_tools::File& file = request.inputfiles(i);
         cout << "Processing file: " << file.name() << std::endl;
         ofstream input_file;
         input_file.open(file.name(), ios::out|ios::binary);
         if (!input_file) {
-            cout << "Unable to open test file for writing" << std::endl;
+            cout << "Unable to open input file for writing" << std::endl;
             return -1;
         }
         input_file << file.content();
         input_file.close();
     }
 
-    ostringstream command;
-    command << request.command() << " " << "> cmd.output 2>&1";
-    system(command.str().c_str());
-    eve_tools::EveTPMResponse response;
+    //sync all the input files.
+    sync();
+
+    //TBD: Propagate the return code all the way to the client, via protobuf.
+    execCmd(request.command());
```

Comment:
How do errors make it back to the caller?
Currently execCmd returns -1 on error, but you don't check that.
But perhaps you should make it return an error string and if that is not empty, then send that string to the caller the same was as the sanitize function does?

Reply:
fixed it in the latest diff.